### PR TITLE
Create Uncss task for deleting unused css

### PR DESCRIPTION
**Pull request by NazarKY**

### CODE REVIEW

**Comment by NazarKY** [comment]
Compressing images
For optimizing images we're using gulp task, but first of all place new images inside source_images directory and run: gulp imagemin

Compressing java-script run command: gulp uglify

Compressing css: 
Put yours styles into _sass folder and run command: gulp uglify

Compressing all assets run command:  gulp minimize

File: css/404.css.map [hunk 1]
```diff
@@ -1 +1 @@
-{"version":3,"file":"404.css","names":[],"mappings":"","sources":["404.scss"],"sourcesContent":["@import url(http://fonts.googleapis.com/css?family=Montserrat:700);\n@import \"pages/error\";"],"sourceRoot":"/source/"}
\ No newline at end of file
+{"version":3,"file":"404.css","names":[],"mappings":"","sources":["404.css"],"sourcesContent":["@import url(http://fonts.googleapis.com/css?family=Montserrat:700);\n\n@import url(http://fonts.googleapis.com/css?family=Montserrat:700);\n\n#error404 {\n  color: #fff;\n  text-align: center;\n  padding-bottom: 60px;\n  background: #2f87b9 url(\"../images/pat_blueprint_light.png\") repeat;\n}\n\n#error404 h1 {\n  font-size: 455px;\n  font-family: 'Montserrat', sans-serif;\n  font-weight: 900;\n  color: #fff;\n  text-shadow: 0 0 30px rgba(0,0,0,0.3);\n}\n\n#error404 .error_message {\n  font-size: 39px;\n  font-weight: 700;\n}\n\n/*# sourceMappingURL=404.css.map */\n"],"sourceRoot":"/source/"}
```
Comment (pftg) [inline css/404.css.map]: we have not changed 404.css, why this map has been updated, also now it's pointed to incorrect source file

File: gulpfile.js [hunk 2]
```diff
@@ -70,3 +75,5 @@ gulp.task('watch', function () {
 gulp.task('jekyll', shell.task('jekyll serve -w'));
 
 gulp.task('default', ['watch', 'jekyll', 'uglify', 'compass']);
+
+gulp.task('minimize', ['imagemin', 'uglify', 'compass', 'uncss']);
```
Comment (pftg) [inline gulpfile.js]: please rename task to `minify`

File: tasks/uncss.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+gulp.task('uncss', function() {
+  gulp.src(['./css/*.css'])
+    .pipe(uncss({
+      html: glob.sync('./_site/**/*.html'),
+      ignore: IGNORE,
+      timeout: 2000
+    }))
+    .pipe(sourcemaps.init())
+    .pipe(cssmin())
+    .pipe(sourcemaps.write('.'))
```
Comment (pftg) [inline tasks/uncss.js]: do not run source maps here, they already has been written in compass

**Comment by pftg** [comment]
@NazarKY please squash commits

File: css/404.css [hunk 4]
```diff
@@ -1,3 +1 @@
-@import url(http://fonts.googleapis.com/css?family=Montserrat:700);@import url(http://fonts.googleapis.com/css?family=Montserrat:700);#error404{color:#fff;text-align:center;padding-bottom:60px;background:#2f87b9 url("../images/pat_blueprint_light.png") repeat}#error404 h1{font-size:455px;font-family:'Montserrat', sans-serif;font-weight:900;color:#fff;text-shadow:0 0 30px rgba(0,0,0,0.3)}#error404 .error_message{font-size:39px;font-weight:700}
-
-/*# sourceMappingURL=404.css.map */
```
Comment (pftg) [inline css/404.css:3]: now we missed `map` ref, please run unss between init source map and finishing.

**Comment by pftg** [comment]
@NazarKY do not forget to squash commits

File: gulpfile.js (hunk 2 above)
Comment (mir4a) [inline gulpfile.js]: @pftg or even better — `gulp min`

Comment (pftg) [inline gulpfile.js]: uglify <-> minify

Comment (mir4a) [inline gulpfile.js]: @pftg `min` is half of `minify`

**Comment by NazarKY** [comment]
Uncss deletes some important css:
Examples of dif page

before uncss
before_uncss_phantomjs diff

after uncss
after_uncss_phantomjs diff

**Comment by NazarKY** [comment]
UnCss works well. It doesn't delete needed css rules, only unused. About comment over - issue was in gemini: before uncss page wasn't loaded appropriately and after deleting unused css (with uncss) page loading was faster - screens were different that was - the reason of difference of screenshots. 

Result of Uncss:

Before uncss:
phantomjs ref

After uncss:
phantomjs current

DIfference 
phantomjs diff

As we see the diff is that text JetThoughts is not on screen but it's only on screenshot. The reason of it is that we need to set more wait time for gemini, after that all will ok.
All report yo can get by this [link](https://drive.google.com/folderview?id=0B9jNBsx5RWYGNjBCZVJSZUJpZTA&usp=sharing)

**Comment by NazarKY** [comment]
@pftg pls review.

**Comment by pftg** [comment]
please rebase?

**Comment by NazarKY** [comment]
TODO:
- [x] deploy via instapusher
- [ ] add report about loading time
- [x] give @mir4a  or @tehhi42 to test

File: tests/gemini_tests/all_pages.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+var HTML_EXTENSION_SIZE = 4
```
Comment (mir4a) [inline tests/gemini_tests/all_pages.js:1]: @NazarKY @pftg нужно будет собрать эталонные скриншоты `gemini gather`

File: .gemini.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+rootUrl: http://localhost:4000/
+screenshotsDir: reports/screenshots
+coverageExclude:
+  - bower_components/
+  - css/
+  - tasks/
+  - images/
+  - js/
+browsers:
+  phantomjs: phantomjs
```
Comment (mir4a) [inline .gemini.yml:10]: @pftg @NazarKY надо попробовать селениум и разные браузеры

File: .gemini.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+rootUrl: http://localhost:4000/
```
Comment (mir4a) [inline .gemini.yml:1]: @pftg @NazarKY на gemini ещё можно писать тесты не на всю страницу а на отдельные элементы, определяя области скриншотов.

**Comment by pftg** [comment]
@mir4a:
1. please place comments for lines if they are for that line, in other case you can place comments for issue
2. all snapshots gathered and stored on google drive

File: tests/gemini_tests/all_pages.js (hunk 5 above)
Comment (pftg) [inline tests/gemini_tests/all_pages.js:1]: why? :laughing:

File: .gemini.yml (hunk 6 above)
Comment (pftg) [inline .gemini.yml:10]: @tehhi42 need to check this issue for different browser if we will find some problems, then will fix in next iteration

File: .gemini.yml (hunk 7 above)
Comment (pftg) [inline .gemini.yml:1]: you mean like this: https://github.com/jetthoughts/jetthoughts.github.io/pull/134/files#diff-b3a9bf5524eaf3d306dcae4720fd151bR5 ?

File: .gemini.yml (hunk 6 above)
Comment (pftg) [inline .gemini.yml:10]: we need for now test others: firefox, safari

**Comment by NazarKY** [comment]
@pftg pls review.

File: _includes/analytics.html [hunk 8]
```diff
@@ -1,4 +1,4 @@
-<script>
+<script async=async>
```
Comment (pftg) [inline _includes/analytics.html]: remove this redundant attrs

**Comment by NazarKY** [comment]
@pftg pls review.

**Pull request merged**